content/blog/angular-components-control-value-accessor/index.md

```diff
@@ -0,0 +1,419 @@
+
+```typescript
+export class ExampleInputComponent implements ControlValueAccessor {
+  // ...
+  constructor(private _changeDetector: ChangeDetectorRef) { }
+   // ...
+ writeValue(value: any) {
+    this.value = value;
+    this._changeDetector.markForCheck();
+  }
+```
+
+Now, when we use a value like `new FormValue('test')` and pass it as `[formControl]` to our component, it will render the correct default value
+
+## `setDisabledState` {#disabled-state}
+
+Implementing the disabled state check is extremely similar to [implementing value writing](#write-value). Simply add a setter, getter, and `setDisabledState` to your component, and you should be good-to-go:
+
+```typescript
+ private _disabled: boolean = false;
+
+  @Input()
+  get disabled(): boolean { return this._disabled; }
+  set disabled(value) {
+    this._disabled = coerceBooleanProperty(value);
+  }
+
+  setDisabledState(isDisabled: boolean) {
+    this.disabled = isDisabled;
+    this._changeDetector.markForCheck();
+  }
+```
+
+Just as we did with value writing, we want to run a `markForCheck` to allow change detection to work as expected when the value is changed from a parent
+
+> It's worth mentioning that unlike the other three methods, this one is entirely optional for implementing a `ControlValueAccessor`. This allows us to disable the component or keep it enabled but is not required for usage with the other methods. `ngModel` and `formControl` will work without this method implemented.
+
+## `registerOnChange` {#register-on-change}
+
+While the previous methods have been implemented in a way that required usage of `markForCheck`, these last two methods are implemented in a bit of a different way. You only need look at the type of the methods on the interface to see as much:
+
+```typescript
+registerOnChange(fn: (value: any) => void);
+```
+
+As you might be able to deduce from the method type, when `registerOnChange` is called, it passes you a function. You'll then want to store this function in your class instance and call it whenever the user changes data.
+
+```typescript
+/** The method to be called to update ngModel */
+_controlValueAccessorChangeFn: (value: any) => void = () => {};
+
+registerOnChange(fn: (value: any) => void) {
+    this._controlValueAccessorChangeFn = fn;
+}
+```
+
+While this code sample shows you how to store the function, but doesn't outline how to call it once stored. You'll want to make sure to call it with the updated value on every update. For example, if you are expecting an `input` to change, you'd want to add it to `(change)` output of the `input`:
```

Comment:
weird sentence structure for that first one. 'but doesn't outline how to call it once stored.' is where it reads oddly.

Reply:
```suggestion
While this code sample shows you how to store the function, it doesn't outline how to call it once stored. You'll want to make sure to call it with the updated value on every update. For example, if you are expecting an `input` to change, you'd want to add it to `(change)` output of the `input`:

```